Elevator/ElevatorControlPanel.lua

```diff
@@ -14,37 +14,144 @@ function drawButton(scr)
   scr.write(scr.text)
 end
 
+function drawFloors()
+	mon.clear()
+	--for each line on screen
+	for i = 0, 10 do -- 10 is line limit of one monitor
+		--only process defined buttons
+		if(floors[i]) then
+			--render
+			drawButton(floors[i])
+		end
+	end
+end
+
+function MonitorEvent(event) -- Select floors etc.
+	if floors[event[4]] then-- Click Y
+		for i = 0, 10 do -- 10 is line limit of one monitor
+			--only process defined buttons
+			if(floors[i]) then
+				floors[i].text = floors[i].text:gsub(">>","")
+			end
+		end
+	
+		--add chevrons to whichever button was hit
+		floors[event[4]].text = ">>"..floors[event[4]].text
+		lastButtonHit = event[4]
+		drawFloors()
+	else
+		return
+    end
+end
+
+function ColorPicker()
+	mon.clear()
+	local _colors = {colors.orange, colors.lightBlue, colors.yellow, colors.pink, colors.cyan, colors.purple, colors.blue, colors.green, colors.red, colors.black}
+	for i = 1, 10 do
+		drawButton(createButton(i,_colors[i], ""))
+	end
+	
+	while true do -- Wait till you get a valid input
+		_,_,_,y = os.pullEvent("monitor_touch")
+		if y then return _colors[y] end
+	end
+end
+
+-- Draws all floors to the monitor, either existing ones, or free spots
+function FloorPicker(onlyExistingFloors)
+	mon.clear()
+	local _colors = {colors.orange, colors.lightBlue, colors.yellow, colors.pink, colors.cyan, colors.purple, colors.blue, colors.green, colors.red, colors.black}
+	local allowed = {}
+	for i = 1, 10 do
+		if floors[i] then
+			if onlyExistingFloors then
+				drawButton(floors[i])
+				allowed[i] = true
+			else -- Occupied spot, but only free floors are needed
+				drawButton(createButton(i,colors.white, "")) -- "None" spot
```

Comment:
I was thinking of using black for missing floors, but we can see which one looks better

Reply:
Font is white, which is why I choose white as reserved background, since white is an illegal background color